src/installer.ts

```diff
@@ -40,7 +40,9 @@ export async function installer(): Promise<void> {
   core.exportVariable("OPAMERRLOGLEN", 0);
   core.exportVariable("OPAMJOBS", numberOfProcessors);
   core.exportVariable("OPAMPRECISETRACKING", 1);
-  core.exportVariable("OPAMSOLVERTIMEOUT", 500);
+  // [todo] remove this line once we unlock opam 2.2
+  // https://github.com/ocaml/opam/issues/3447
+  core.exportVariable("OPAMSOLVERTIMEOUT", 0);
```

Comment:
I don't think this should be unlimited - a bigger value, perhaps, but otherwise this runs a risk of just maxing the timeout on the GitHub actions runner.

Reply:
Well, depending on the size of the project, this still has the potential to fail. What do you think of using the options of the exec function to be unlimited only when executing depext?

Reply:
Using the env option seems somewhat more promising.

```ts
export interface ExecOptions {
    /** optional working directory.  defaults to current */
    cwd?: string;
    /** optional envvar dictionary.  defaults to current process's env */
    env?: {
        [key: string]: string;
    };
    /** optional.  defaults to false */
    silent?: boolean;
    /** optional out stream to use. Defaults to process.stdout */
    outStream?: stream.Writable;
    /** optional err stream to use. Defaults to process.stderr */
    errStream?: stream.Writable;
    /** optional. whether to skip quoting/escaping arguments if needed.  defaults to false. */
    windowsVerbatimArguments?: boolean;
    /** optional.  whether to fail if output to stderr.  defaults to false */
    failOnStdErr?: boolean;
    /** optional.  defaults to failing on non zero.  ignore will not fail leaving it up to the caller */
    ignoreReturnCode?: boolean;
    /** optional. How long in ms to wait for STDIO streams to close after the exit event of the process before terminating. defaults to 10000 */
    delay?: number;
    /** optional. input to write to the process on STDIN. */
    input?: Buffer;
    /** optional. Listeners for output. Callback functions that will be called on these events */
    listeners?: ExecListeners;
}
```

Reply:
Ah - do you mean only setting `OPAMSOLVERTIMEOUT` if the user of setup-ocaml hasn't set it in their workflow? That seems good, yes.